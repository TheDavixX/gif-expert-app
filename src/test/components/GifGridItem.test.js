import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas con <CounterApp />', ()=>{
    const title = "Esto es un titulo";
    const url = "https://localhost/logo.png";
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

    test('debe mostrar <GifGridItem /> correctamente', () => {  

        expect( wrapper ).toMatchSnapshot();
    
    });

    test('debe tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        
        expect(p.text().trim()).toBe(title);

    });

    test('debe de tener la imagen igual al url y alt de las props', () => {
        
        const img = wrapper.find('img');

        expect( img.props().src ).toBe( url );
        expect( img.props().alt ).toBe( title );

    });

    test('debe de tener la clase animate__rubberBand', () => {
        
        const img = wrapper.find('div');
        const classes = img.props().className;

        expect( classes.includes('animate__rubberBand') ).toBe( true );

    });
});