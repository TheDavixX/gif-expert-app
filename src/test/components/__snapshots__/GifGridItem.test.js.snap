// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <CounterApp /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Esto es un titulo"
    src="https://localhost/logo.png"
  />
  <p>
     
    Esto es un titulo
  </p>
</div>
`;
